src/table/table.component.ts

```diff
@@ -837,13 +837,25 @@ export class Table implements AfterViewInit, OnDestroy {
 	}
 
 	doSort(index: number) {
+		const currentHeader = this.model.header[index];
 		if (this.sort.observers.length === 0) {
 			// no sort provided so do the simple sort
-			if (this.model.header[index].sorted) {
-				// if already sorted flip sorting direction
-				this.model.header[index].ascending = this.model.header[index].descending;
+			if (!currentHeader.sorted) {
+			// add origin sort for newData
+			this.model.data.forEach((rowItem, idx) => {
+				rowItem.forEach(colItem => {
+					colItem.originIdx = idx;
+				});
+			});
+			currentHeader.ascending = true;
+				this.model.sort(index);
+			} else if (currentHeader.sorted && currentHeader.ascending) {
+				currentHeader.ascending = false;
+				currentHeader.descending = true;
+				this.model.sort(index);
+			} else {
+				this.model.resetSort();
```

Comment:
```suggestion
			if (!currentHeader.sorted) {
				// add origin sort for newData
				this.model.data.forEach((rowItem, index) => {
					rowItem.forEach(colItem => {
						colItem.index = index;
					});
				});
				currentHeader.ascending = true;
				this.model.sort(index);
			} else if(currentHeader.sorted && currentHeader.ascending) {
				// if already sorted flip sorting direction
				currentHeader.descending = true;
				this.model.sort(index);
			} else {
				this.model.resetSort();
```

Reply:
No need to set ascending to false since headerItem class uses `ascending` for both ascending (true) & descending (false). Instead of using `idx` just use index or `i`.